### DISCUSSION TITLE
CSRF Block on fetch POST within iFrame in Pimcore (/admin)

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#17721] [Q&A / Support]
I’m encountering an issue where Pimcore is detecting fetch POST requests from an iFrame to the /admin route as a potential CSRF (Cross-Site Request Forgery) attack.

GET requests work fine within the iFrame, but when making a POST request, it triggers a CSRF error.
It seems that Pimcore’s security policies are blocking POST requests in iFrames, recognizing them as potentially malicious.

I’d appreciate guidance on how to resolve this issue or suggestions on best practices for handling POST requests from iFrames in Pimcore.

![image](https://github.com/user-attachments/assets/2665febb-3ccb-44d3-b06d-3af36c824348)

**Comment by AlephLau** [comment]
@mattamon can you help me with this pls?

**Reply by AlephLau** [reply]
@fashxp Could you give me a little help please?

**Reply by fashxp** [reply]
sorry, not out of my head ... would need to investigate my own... 
but maybe it is again related to the /admin route, where Pimcore eventually centrally checks for CSRF tokens on POST requests.
so, probably you need to send the CSRF token with the request or change the route to something else...

**Reply by AlephLau** [reply]
I want to continue using the /admin route in the requests I make from the pimcore admin, but I have run into this problem, I have tried to send a csrf token but the error continues to appear, I don't know if I should give it a name in cpontrI believe, thank you anyway, I will continue investigating
